src/Chapter2.hs

```diff
@@ -841,7 +850,11 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n l = 
+  let llen = length l
+      ls = take (2 * llen) (cycle l)
+  in take llen (drop n ls)
```

Comment:
That's a correct solution! Though, it does a little bit more than work when the given number is big and the list is small. You will cycle through the list a lot. Otherwise, if `n > 2 * llen`, `drop` returns an empty list.

Maybe you can think of a way to optimize this and handle these special cases :slightly_smiling_face:  :slightly_smiling_face: